### DISCUSSION TITLE
Is that possible to create an official Avalonia runtime redistribution?

### DISCUSSION THREAD

**Discussion by laolarou726** [AvaloniaUI/Avalonia#13582] [Ideas]
Just like DOTNET's runtime installer, Avalonia can also make a redistribution for its runtime. This will significantly reduce the size of the executable, and also increase the reusability of the runtime files (imagine that there are multiple Avalonia applications on the computer, that will be a waste of disk space).

**Comment by laolarou726** [comment]
<img width="747" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/25716486/02bb10d4-e3b5-403e-90d1-b64689ea2dcf">

**Reply by maxkatz6** [reply]
From this screenshot, SkiaSharp runtime redistribution makes more sense. 
But either way, I think (?), GAC assemblies are still supported on Windows. I.e. you can install some dlls into the system folder, and it will be used if this library is missing from the app folder. And I don't think it's possible on other OS.

**Reply by thevortexcloud** [reply]
GAC assemblies do work on Windows. But they wont work by default. You need to manually add the GAC path to the assembly search path per project in the project file (I can't remember how to do it off the top of my head though, and it's hard to find examples for it). They don't want people using it without a good reason. Not to mention the entire concept of the GAC has been obsoleted. It's really also only meant to be used for .NET Framework assemblies.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/global-assembly-cache-apis-obsolete

Anyway, the GAC is still supported for .NET Framework.

Nuget already caches packages which is meant to serve most of the same purpose, at least during development. 

Relying on shared libraries for deployment generally leads to dependency hell. Which a problem Linux users regularly have and also why every distro needs to recompile every native app for their package manager, as sharing apps across distros is in no way guaranteed to work.
